Pear/Profile/ProfileViewController.swift

```diff
@@ -0,0 +1,284 @@
+        }
+
+        view.addSubview(reachOutButton)
+    }
+
+    private func setupConstraints() {
+        let reachOutPadding = LayoutHelper.shared.getCustomVerticalPadding(size: 70)
+        let meetupPadding = 24
+        let meetupWidth: CGFloat = 319
+
+        meetupStatusView?.snp.makeConstraints { make in
+            make.top.equalToSuperview().offset(meetupPadding)
+            make.leading.equalTo(view.safeAreaLayoutGuide).inset(meetupPadding)
+            make.width.equalTo(meetupWidth)
+            make.height.equalTo(meetupStatusView?.getRecommendedHeight(for: meetupWidth) ?? 0)
+        }
+
+        if let meetupStatusView = meetupStatusView {
+            profileTableView.snp.makeConstraints { make in
+                make.leading.trailing.bottom.equalToSuperview()
+                make.top.equalTo(meetupStatusView.snp.bottom).offset(8)
+            }
+        } else {
+            profileTableView.snp.makeConstraints { make in
+                make.edges.equalToSuperview()
+            }
+        }
+
+        reachOutButton.snp.makeConstraints { make in
+            make.bottom.equalTo(view.safeAreaLayoutGuide).inset(reachOutPadding)
+            make.centerX.equalToSuperview()
+            make.size.equalTo(reachOutButtonSize)
+        }
+    }
+
+    @objc private func reachOutPressed() {
+        switch type {
+        case .match:
+            guard let pair = pair, let match = match, let user = user else { return }
+            let schedulingVC: SchedulingTimeViewController
+
+            switch chatStatus {
+            case .planning, .noResponses:
+                schedulingVC = SchedulingTimeViewController(for: .confirming, user: user, pair: pair, match: match)
+                navigationController?.pushViewController(schedulingVC, animated: true)
+            case .respondingTo:
+                print("thsi is my match", match)
+                schedulingVC = SchedulingTimeViewController(for: .choosing, user: user, pair: pair, match: match)
+                navigationController?.pushViewController(schedulingVC, animated: true)
+            default:
+                URLScheme.openGmail(to: "\(pair.netID)@cornell.edu", subject: "Hello from Pear!")
+            }
+        case .user:
+            guard let pair = pair else { return }
+            URLScheme.openGmail(to: "\(pair.netID)@cornell.edu", subject: "Hello from Pear!")
+        }
+    }
+
+    @objc func backPressed() {
+        navigationController?.popViewController(animated: true)
+    }
+}
+
+extension ProfileViewController: UITableViewDelegate {}
+
+extension ProfileViewController: UITableViewDataSource {
+    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        return profileSections.count
+    }
+
+    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        guard let pair = pair else { return UITableViewCell() }
```

Comment:
This is a VERY big class, and this guard right here indicates that you can definitely break it up. I recommend that you refactor: maybe create a view that takes a `pair` as a _not_ optional parameter. This will make the code more concise, organized, readable, and maintainable. _~~decoupling~_~